snapcraft/internal/pluginhandler/__init__.py

```diff
@@ -517,10 +517,6 @@ def clean_pull(self):
         if self.is_clean(steps.PULL):
             return
 
-        # Remove stage-packages cache (where stage packages are fetched)
```

Comment:
Does this mean that if you remove a stage-packages entry from a list that would remove the need to bring set packages in that those packages would still be staged?

Reply:
the cache directory is common for everything now (~/.cache/snapcraft/download).  install_stage_packages() is the one that extracts the list of packages.  So assuming the part was cleaned (that is, it detects the change in stage-packages and rebuilds), that stage package is never extracted.